### DISCUSSION TITLE
How do I create a Map<String, Integer> in Json Schema

### DISCUSSION THREAD

**Discussion by mathan-mohan91** [json-schema-org/community#651] [Q&A] [answered]
Hey all,
I had a doubt on creating a Map<String, Integer> where key needs to be validated against an enum and integer needs to be validated for minimum and maximum. 
Is there a way to achieve this in Json Schema? 

Tried the below but it doesn't to work as it's not validating against the enum values. 

```
"levels": {
      "type": "object",
      "enum": ["En_515"],
      "additionalProperties": {
        "$ref": "#/definitions/mapInt"
      }
    },
    "mapInt": {
      "type": "object",
      "additionalProperties": {"type": "integer"}
    },
```
Tested with below Json and it doesn't work. 
```
"levels": {
      "Un_515": 3
    }
```

**Comment by jdesrosiers** [comment, marked as answer]
The are a few approaches. I'd recommend this one.

```json
{
  "propertyNames": {
    "enum": ["En_515"]
  },
  "patternProperties": {
    "": { "type": "integer" }
  }
}
```

**Reply by mathan-mohan91** [reply]
Thanks for the response but it doesn't seem to work. Not sure if I'm missing something here. 

```
    "levels": {
      "type": "object",
      "propertyNames": {
        "enum": [
          "En_515"
        ]
      },
      "patternProperties": {
        "": {
          "type": "integer"
        }
      }
    }
```

I have provided a value not in enum

```
    "levels": {
      "SI": 100
    }
```
Also is it possible to put a min and max on integer?

**Reply by mathan-mohan91** [reply]
Thanks, it seems to work.
